docs/markdown/Rust-module.md

```diff
@@ -33,3 +33,31 @@ that automatically.
 
 Additional, test only dependencies may be passed via the dependencies
 argument.
+
+### bindgen(input: str, output: str, *, include_directories: []include_directories, c_args: []string, args: []string)
+
+This function wraps bindgen to simplify creating rust bindings around C
+libraries. It takes two positional arguments; `input`, which is the
+path to the header to wrap, and `output`, which is the name of the file to
+write.
+
+It additionally takes the following keyword arguments:
+
+- include_directories — A list of `include_directories` objects, these are
+  passed to clang as `-I` arguments
+- c_args — A list of string arguments to pass to clang untouched
+- args — A list of string arguments to pass to `bindgen` untouched.
+
+```meson
+rust = import('unstable-rust')
+
+inc = include_directories('..'¸ '../../foo')
+
+generated = rust.bindgen(
+    'myheader.h',
+    'generated.rs',
+    include_directories : [inc, include_directories('foo'),
```

Comment:
Question: In my use case, the C header also depends on generated headers (i.e. their directory is inside build dir). What would be a way to add them here?

Otherwise, the interface described here looks fine to me.

Reply:
You're right, we do need to handle generated headers.